### PR TITLE
feat: add test-validator --stop flag

### PR DESCRIPTION
**Pull request by SwenSchaeferjohann**
make it easy for users to stop the test-validator and related background processes

### CODE REVIEW

File: cli/src/commands/test-validator/index.ts [hunk 1]
```diff
@@ -80,29 +80,43 @@ class SetupCommand extends Command {
       required: false,
       default: "127.0.0.1",
     }),
+    stop: Flags.boolean({
+      description: "Stops the test validator and dependent processes.",
+      required: false,
+      default: false,
```
Comment (sergeytimoshin) [inline cli/src/commands/test-validator/index.ts:86]: It's not critical, but I think it makes sense to add the `exclusive: [<all other available flags>]` parameter or even make this a standalone command.

File: cli/src/utils/initTestEnv.ts [hunk 2]
```diff
@@ -17,9 +17,45 @@ import {
...
+      condition: true,
+      killFunction: killTestValidator,
+    },
+  ];
+
+  for (const process of processesToKill) {
+    if (process.condition) {
+      try {
+        if (process.killFunction) {
+          await process.killFunction();
```
Comment (sergeytimoshin) [inline cli/src/utils/initTestEnv.ts:48]: We can use array of promises here to stop all the processes at once.

File: cli/src/commands/test-validator/index.ts (hunk 1 above)
Comment (SwenSchaeferjohann) [reply cli/src/commands/test-validator/index.ts:86]: user can pass eg `--stop --skip-prover` to keep those alive

Comment (SwenSchaeferjohann) [reply cli/src/commands/test-validator/index.ts:86]: gonna add to 'description' for clarity

**Review by sergeytimoshin — APPROVED** [review]

**Pull request merged**